code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,123 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	anchored = FALSE
+	density = TRUE
+	use_power = POWER_USE_OFF
+	var/obj/item/cell/cell = /obj/item/cell/high
+	var/on = FALSE
+	var/drainage = TRUE
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+/obj/machinery/dehumidifier/Initialize()
+	if(ispath(cell))
+		cell = new cell(src)
+	. = ..()
+	update_icon()
+
+/obj/machinery/dehumidifier/on_update_icon(var/rebuild_overlay = 0)
+	icon_state = "dhum-[on ? "on" : "off"]"
+	if(rebuild_overlay)
+		overlays.Cut()
+		if(panel_open)
+			overlays.Add("dhum-open")
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..(user)
+	if (!.)
+		return
+
+	to_chat(user, "The dehumidifier is [on ? "on" : "off"] and the hatch is [panel_open ? "open" : "closed"].")
+	if(panel_open)
+		to_chat(user, "The power cell is [cell ? "installed" : "missing"].")
+	else
+		to_chat(user, "The charge meter reads [cell ? round(cell.percent(),1) : 0]%")
+
+/obj/machinery/dehumidifier/emp_act(severity)
+	if(cell)
+		cell.emp_act(severity)
+	..(severity)
+
+/obj/machinery/dehumidifier/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/cell))
+		if(panel_open)
+			if(cell)
+				to_chat(user, "There is already a power cell inside.")
+				return
+			else
+				// insert cell
+				if(!user.unEquip(I, src))
+					return
+				cell = I
+				I.forceMove(src)
+				user.visible_message(SPAN_NOTICE("[user] inserts a power cell into [src]"), SPAN_NOTICE("You insert the power cell into [src]."))
+				power_change()
+				return
+		else
+			to_chat(user, "The hatch must be open to insert a power cell.")
+			return
+	else if(isScrewdriver(I))
+		panel_open = !panel_open
```

Comment:
Use construct decls to handle all of your construct stuff. That + using the battery component means you can remove 80% of the code in this file.